h2o-core/src/main/java/water/AutoBuffer.java

```diff
@@ -1475,12 +1475,20 @@ public static Object javaSerializeReadPojo(byte [] bytes) {
     try {
       return new ObjectInputStream(new ByteArrayInputStream(bytes)).readObject();
     } catch (IOException e) {
-      String className = new String(bytes, 8, Math.min(40, Math.max(3, bytes[7])));
+      String className = nameOfClass(bytes);
       throw Log.throwErr(new RuntimeException("Failed to deserialize " + className, e));
     } catch (ClassNotFoundException e) {
       throw Log.throwErr(e);
     }
   }
+
+  static String nameOfClass(byte[] bytes) {
+    assert bytes != null : "pojo bytes can't be null";
```

Comment:
Idea: we can return "Unknown" if there is any exception during the call of this method

assertions will not be applied during production deployment and it might make our logs less readable if an unexpected exception suppresses the causative exception (the suppressed exception will still be available in Java 7+ but I think it still make sense to avoid it if possible)